src/common/types/list_segment.cpp

```diff
@@ -668,7 +619,6 @@ void GetSegmentDataFunctions(ListSegmentFunctions &functions, const LogicalType
 		functions.write_data = WriteDataToStructSegment;
 		functions.read_data = ReadDataFromStructSegment;
 		functions.copy_data = CopyDataFromStructSegment;
-		functions.destroy = DestroyStructSegment;
```

Comment:
Is this required for the repartitioning to work? If so - should all aggregates with destructors have their destructors removed in order for this to work? Can we perhaps add large tests with `verify_external` for all aggregates with destructors (including e.g. `min/max/first` on non-inlined string values, `string_agg`, `histogram`, etc).